### PR TITLE
Add aarch64 wheel build support

### PR DESCRIPTION
**Pull request by odidev**
Added aarch64 wheel build support.
@FrancescAlted Could you please review this PR?

### CODE REVIEW

File: .github/workflows/cibuildwheels.yml [hunk 1]
```diff
@@ -21,12 +21,12 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        arch: [x86, amd64]
+        arch: [auto, aarch64]
```
Comment (FrancescAlted) [inline .github/workflows/cibuildwheels.yml:24]: What `auto` means here?  Why `x86` and `amd64` are gone?

File: .github/workflows/cibuildwheels.yml [hunk 2]
```diff
@@ -54,7 +58,7 @@ jobs:
           python -m pip install -r requirements-tests.txt
 
       - name: Build wheels (Windows x86)
-        if: ${{ runner.os == 'Windows' && matrix.arch == 'x86'}}
+        if: ${{ runner.os == 'Windows' }}
```
Comment (FrancescAlted) [inline .github/workflows/cibuildwheels.yml:61]: Why you are removing the `matrix.arch == 'x86'` condition?  Is that related with `auto`?

File: .github/workflows/cibuildwheels.yml [hunk 3]
```diff
@@ -72,7 +76,7 @@ jobs:
           arch: amd64
 
       - name: Build wheels (Windows / amd64)
-        if: ${{ runner.os == 'Windows' && matrix.arch == 'amd64'}}
+        if: ${{ runner.os == 'Windows' }}
```
Comment (FrancescAlted) [inline .github/workflows/cibuildwheels.yml:79]: Ditto.

**Review by FrancescAlted — COMMENTED** [review]
Sorry for the delay on this.  I got diverted and forgot about this.

I would appreciate if you can shed some light on my comments.  Thanks in advance.

File: .github/workflows/cibuildwheels.yml (hunk 1 above)
Comment (hirnimeshrampuresoftware) [reply .github/workflows/cibuildwheels.yml:24]: Here `auto` takes all the available architecture for the given OS. Hence it will automatically take `x86` and `amd64`. Also `aarch64` is not available under `auto`, hence it is provided separately.

File: .github/workflows/cibuildwheels.yml (hunk 2 above)
Comment (hirnimeshrampuresoftware) [reply .github/workflows/cibuildwheels.yml:61]: Since `x86` and `amd64` are replaced with `auto`, this condition is removed. Also, `CIBW_BUILD` is provided in the `env` which will build the required wheel.

File: .github/workflows/cibuildwheels.yml (hunk 1 above)
Comment (odidev) [reply .github/workflows/cibuildwheels.yml:24]: Here `auto` takes all the available architecture for the given OS. Hence it will automatically take `x86` and `amd64`. Also `aarch64` is not available under `auto`, hence it is provided separately.

File: .github/workflows/cibuildwheels.yml (hunk 2 above)
Comment (odidev) [reply .github/workflows/cibuildwheels.yml:61]: Since `x86` and `amd64` are replaced with `auto`, this condition is removed. Also, `CIBW_BUILD` is provided in the `env` which will build the required wheel.

File: .github/workflows/cibuildwheels.yml (hunk 1 above)
Comment (FrancescAlted) [reply .github/workflows/cibuildwheels.yml:24]: Got it.

File: .github/workflows/cibuildwheels.yml (hunk 2 above)
Comment (FrancescAlted) [reply .github/workflows/cibuildwheels.yml:61]: Understood.  So I'd say that the section below ('Build wheels (Windows / amd64)') is redundant.

Comment (FrancescAlted) [reply .github/workflows/cibuildwheels.yml:61]: If I understand this correctly, this also means that this should be producing wheels for `aarch64` arch too.

Comment (odidev) [reply .github/workflows/cibuildwheels.yml:61]: No, It produces only for the supported architectures of a build machine.

Comment (odidev) [reply .github/workflows/cibuildwheels.yml:61]: > Understood. So I'd say that the section below ('Build wheels (Windows / amd64)') is redundant.

Yes, this can be removed.

Comment (FrancescAlted) [reply .github/workflows/cibuildwheels.yml:61]: Thanks!

File: .github/workflows/cibuildwheels.yml (hunk 3 above)
Comment (FrancescAlted) [reply .github/workflows/cibuildwheels.yml:79]: This section would be removed.

**Comment by FrancescAlted** [comment]
Due to reasons explained in https://github.com/Blosc/python-blosc2/pull/47#issuecomment-1225633013, the aarch64 wheels will be disabled for now.
